src/commands/lambda/__tests__/flare.test.ts

```diff
@@ -15,7 +67,192 @@ describe('lambda flare', () => {
+    process.env[API_KEY_ENV_VAR] = 'test-api-key'
+    code = await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'test-region', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(code).toBe(0)
+  })
+
+  it('prints error when no case ID specified', async () => {
+    const cli = makeCli()
+    const context = createMockContext()
+    const code = await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(code).toBe(1)
+    const output = context.stderr.toString()
+    expect(output).toMatchSnapshot()
+  })
+
+  it('prints error when no email specified', async () => {
+    const cli = makeCli()
+    const context = createMockContext()
+    const code = await cli.run(['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123'], context as any)
+    expect(code).toBe(1)
+    const output = context.stderr.toString()
+    expect(output).toMatchSnapshot()
+  })
+
+  it('requests AWS credentials when none are found and logs output correctly', async () => {
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    const output = context.stdout.toString()
+    expect(output).toMatchSnapshot()
+    expect(requestAWSCredentials).toHaveBeenCalled()
+  })
+
+  it('runs successfully with all required options specified', async () => {
+    const cli = makeCli()
+    const context = createMockContext()
+    const code = await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(code).toBe(0)
+  })
+
+  it('successfully adds zip file to FormData', async () => {
+    const appendSpy = jest.spyOn(FormData.prototype, 'append')
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(appendSpy).toHaveBeenCalledWith('flare_file', expect.anything())
+    appendSpy.mockRestore()
+  })
+
+  it('successfully sends request to Datadog', async () => {
+    const postSpy = jest.spyOn(axios, 'post').mockResolvedValue({status: 200})
+    const cli = makeCli()
+    const context = createMockContext()
+    await cli.run(
+      ['lambda', 'flare', '-f', 'func', '-r', 'us-west-2', '-c', '123', '-e', 'test@test.com'],
+      context as any
+    )
+    expect(postSpy).toHaveBeenCalledWith(
+      expect.anything(),
+      expect.anything(),
+      expect.objectContaining({
+        headers: expect.objectContaining({
+          'DD-API-KEY': 'test-api-key',
+        }) as Record<string, string>,
+      })
+    )
+    const output = context.stdout.toString()
+    expect(output).toMatchSnapshot()
+    postSpy.mockRestore()
+  })
```

Comment:
I like how we are separating what we test, but I found it a little bit confusing because the first and last test are interchangable in the sense that they both throw the same data? How could we improve this?